### PR TITLE
chart-expense-sprint5

### PR DESCRIPTION
**Pull request by Manish-Srans**
Update Hostel Controller, modify doughnut chart API

### CODE REVIEW

File: .classpath [hunk 1]
```diff
@@ -13,19 +13,37 @@
 	</classpathentry>
 	<classpathentry kind="src" output="target/test-classes" path="src/test/java">
```
Comment (srans-admin) [inline .classpath:14]: This file should not push

File: .factorypath [hunk 2]
```diff
@@ -0,0 +1,68 @@
+<factorypath>
+    <factorypathentry kind="VARJAR" id="M2_REPO/org/springframework/boot/spring-boot-starter-data-jpa/2.0.5.RELEASE/spring-boot-starter-data-jpa-2.0.5.RELEASE.jar" enabled="true" runInBatchMode="false"/>
```
Comment (srans-admin) [inline .factorypath:2]: This file should not pushed

File: src/main/java/com/srans/nestserver/controller/ChartController.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+@CrossOrigin(origins = "http://localhost:4200", allowedHeaders = "*")
+@RequestMapping("api/v1")
+@RestController
+public class ChartController {
+
+	@Autowired
+	private BedRepository bedRepository;
+
+	@GetMapping("hostels/{id}/singlebedinfo")
+	public List<Object> getHostelInfo(@PathVariable(value = "id") Long hostelId) {
```
Comment (srans-admin) [inline src/main/java/com/srans/nestserver/controller/ChartController.java]: Some how we need to manage this in generic way

File: src/main/java/com/srans/nestserver/controller/ChartController.java [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+		return Arrays.asList(
+
+				bedRepository.totalNumberOfBeds(hostelId), bedRepository.totalEmptyBeds(hostelId),
+				bedRepository.totalFilledBeds(hostelId));
+	}
+
+	@GetMapping("hostels/{id}/doublebedinfo")
+	public List<Object> getHostelInfo1(@PathVariable(value = "id") Long hostelId) {
```
Comment (srans-admin) [inline src/main/java/com/srans/nestserver/controller/ChartController.java]: same as above

File: src/main/java/com/srans/nestserver/controller/ChartController.java [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+		return Arrays.asList(
+
+				bedRepository.totalNumberOfBeds1(hostelId), bedRepository.totalEmptyBeds1(hostelId),
+				bedRepository.totalFilledBeds1(hostelId));
+	}
+
+	@GetMapping("hostels/{id}/triplebedinfo")
+	public List<Object> getHostelInfo2(@PathVariable(value = "id") Long hostelId) {
```
Comment (srans-admin) [inline src/main/java/com/srans/nestserver/controller/ChartController.java]: Here also returing an array it does not mean what JSON it is, some what clearly plan the API design with some standards

File: src/main/java/com/srans/nestserver/controller/ChartController.java [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+		return Arrays.asList(
+
+				bedRepository.totalNumberOfBeds2(hostelId), bedRepository.totalEmptyBeds2(hostelId),
+				bedRepository.totalFilledBeds2(hostelId));
+	}
+
+	@GetMapping("hostels/{id}/miscbedinfo")
```
Comment (srans-admin) [inline src/main/java/com/srans/nestserver/controller/ChartController.java]: We need to remove misc, beacause there is no meaning of misc is

File: src/main/java/com/srans/nestserver/controller/ChartController.java [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+	public List<Object> getHostelInfo3(@PathVariable(value = "id") Long hostelId) {
```
Comment (srans-admin) [inline src/main/java/com/srans/nestserver/controller/ChartController.java]: method names should be appropriate and readable definitions

File: src/main/java/com/srans/nestserver/controller/ChartController.java [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+		return Arrays.asList(
+
+				bedRepository.totalNumberOfBeds3(hostelId), bedRepository.totalEmptyBeds3(hostelId),
+				bedRepository.totalFilledBeds3(hostelId));
+	}
+	
+	@GetMapping(value= "hostels/{id}/chart")
```
Comment (srans-admin) [inline src/main/java/com/srans/nestserver/controller/ChartController.java]: chart is UI functionality, for backend either it is kind of single/double/ .. bed kind of information, rename the method to show generic implementation

File: src/main/java/com/srans/nestserver/controller/ExpensesController.java [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+	@Autowired
+	private TenantRepository tenantRepository;
+
+	@GetMapping("/expenses")
+	public List<Expense> getAllExpenses() {
+
+		return expensesRepository.findAll();
+	}
+
+	/*
```
Comment (srans-admin) [inline src/main/java/com/srans/nestserver/controller/ExpensesController.java]: Documentation comment starts as /** , you inserted normal comment, change that

File: src/main/java/com/srans/nestserver/controller/ExpensesController.java [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+	 * @GetMapping("/expenses/{id}") public ResponseEntity<Expense>
+	 * getExpensesById(@PathVariable(value = "id") Long ExpensesId) throws
+	 * ResourceNotFoundException { Expense expenses =
+	 * expensesRepository.findById(ExpensesId) .orElseThrow(() -> new
+	 * ResourceNotFoundException(" Expenses not found for this id :: " +
+	 * ExpensesId)); return ResponseEntity.ok().body(expenses); }
+	 */
+
+	@GetMapping("expenses/hostelName")
+	public List<String> findAll() {
```
Comment (srans-admin) [inline src/main/java/com/srans/nestserver/controller/ExpensesController.java]: ExpensesController , why hard coded hostelName, no meaning here

File: src/main/java/com/srans/nestserver/controller/ExpensesController.java [hunk 11]
```diff
@@ -0,0 +1,102 @@
...
+
+	@Autowired
+	private HostelRepository hostelRepository;
+
+	@Autowired
+	private TenantRepository tenantRepository;
+
+	@GetMapping("/expenses")
+	public List<Expense> getAllExpenses() {
+
```
Comment (srans-admin) [inline src/main/java/com/srans/nestserver/controller/ExpensesController.java:52]: Why logging is not implemented here? without logging how can we debug ?

File: src/main/java/com/srans/nestserver/controller/ExpensesController.java [hunk 12]
```diff
@@ -0,0 +1,102 @@
...
+	/*
+	 * @GetMapping("/expenses/{id}") public ResponseEntity<Expense>
+	 * getExpensesById(@PathVariable(value = "id") Long ExpensesId) throws
+	 * ResourceNotFoundException { Expense expenses =
+	 * expensesRepository.findById(ExpensesId) .orElseThrow(() -> new
+	 * ResourceNotFoundException(" Expenses not found for this id :: " +
+	 * ExpensesId)); return ResponseEntity.ok().body(expenses); }
+	 */
+
+	@GetMapping("expenses/hostelName")
```
Comment (srans-admin) [inline src/main/java/com/srans/nestserver/controller/ExpensesController.java]: API design - no stanadad

File: src/main/java/com/srans/nestserver/controller/ExpensesTypeController.java [hunk 13]
```diff
@@ -0,0 +1,79 @@
...
+@RequestMapping("/api/v1")
+public class ExpensesTypeController {
+	Logger logger = LoggerFactory.getLogger(ExpensesTypeController.class);
+	@Autowired
+	private ExpenseTypeRepository expensesTypeRepository;
+
+	@Autowired
+	private HostelRepository hostelRepository;
+
+	@GetMapping("/expensestype")
```
Comment (srans-admin) [inline src/main/java/com/srans/nestserver/controller/ExpensesTypeController.java]: use - if two words are there

File: src/main/java/com/srans/nestserver/controller/ExpensesTypeController.java [hunk 14]
```diff
@@ -0,0 +1,79 @@
...
+	public List<ExpensesType> getAllExpensesType() {
+
+		return expensesTypeRepository.findAll();
+	}
+
+	@PostMapping("/expensestype/{id}")
```
Comment (srans-admin) [inline src/main/java/com/srans/nestserver/controller/ExpensesTypeController.java]: use -

File: src/main/java/com/srans/nestserver/controller/ExpensesTypeController.java [hunk 15]
```diff
@@ -0,0 +1,79 @@
...
+	public ExpensesType createExpenseType(@PathVariable(value = "id") Long id,
+			@Valid @RequestBody ExpensesType expensestype) {
```
Comment (srans-admin) [inline src/main/java/com/srans/nestserver/controller/ExpensesTypeController.java]: where is logging

File: src/main/java/com/srans/nestserver/controller/HostelController.java [hunk 16]
```diff
@@ -17,59 +22,271 @@
...
+	@PostMapping("/hostels")
+	public Hostel saveHostel(@Valid @RequestBody Hostel hostel) throws NSException {
+
+		logger.info("IN::POST::/hostels::saveHostel::" + hostel);
+
+		Hostel responseHostel = hostelRepository.save(hostel);
+
+		// SAVE Database stuff here
+
+		responseHostel.getfloors().forEach(floor -> {
```
Comment (srans-admin) [inline src/main/java/com/srans/nestserver/controller/HostelController.java:94]: SaveHostel , looks good as you saved all sub equivalent associated objects been saved, i dont see same structure of implementation followed on other /hostels related API calls, this should be consistent across all

File: src/main/java/com/srans/nestserver/controller/HostelController.java [hunk 17]
```diff
@@ -17,59 +22,271 @@
...
-				.orElseThrow(() -> new ResourceNotFoundException("Hostel not found for this id :: " + employeeId));
+	public ResponseEntity<?> deleteHostel(@PathVariable Long id) {
+		return hostelRepository.findById(id).map(hostel -> {
+			hostelRepository.delete(hostel);
+			return ResponseEntity.ok().build();
+		}).orElseThrow(() -> new ResourceNotFoundException("HostelId " + id + " not found"));
+	}
+
+	@GetMapping("/hostels/{id}/floor")
+	public List<Floor> getAllFloorsByHostelid(@PathVariable(value = "id") Long id) {
```
Comment (srans-admin) [inline src/main/java/com/srans/nestserver/controller/HostelController.java:157]: Cross check typos

File: src/main/java/com/srans/nestserver/controller/HostelController.java [hunk 18]
```diff
@@ -17,59 +22,271 @@
...
+		logger.info("All Floor Getting");
+		return floorRepository.findByHostelId(id);
+	}
+
+	@GetMapping("/hostels/{id}/floor/{floor_id}")
```
Comment (srans-admin) [inline src/main/java/com/srans/nestserver/controller/HostelController.java:162]: java terminology - Not floor_id, use floorId

File: src/main/java/com/srans/nestserver/controller/HostelController.java [hunk 19]
```diff
@@ -17,59 +22,271 @@
...
+			floor.setFloorName(floorRequest.getFloorName());
+			floor.setDescription(floorRequest.getDescription());
+			return floorRepository.save(floor);
+		}).orElseThrow(() -> new ResourceNotFoundException("floorId " + floor_id + "not found"));
+
+	}
+
+	@DeleteMapping("/hostels/{id}/floors/{floor_id}")
+	public ResponseEntity<?> deleteFloor(@PathVariable(value = "id") Long id,
+
```
Comment (srans-admin) [inline src/main/java/com/srans/nestserver/controller/HostelController.java:205]: Logging for entire class ?

File: src/main/java/com/srans/nestserver/controller/HostelController.java [hunk 20]
```diff
@@ -17,59 +22,271 @@
...
+		return Arrays.asList(
+
+				hostelRepository.hostelName(hostelId), hostelRepository.numOfFloor(hostelId),
+				hostelRepository.hostelType(hostelId), hostelRepository.hostelAddress(hostelId),
+				roomRepository.countRoomByHostelId(hostelId), roomRepository.countSingleSharing(hostelId),
+				roomRepository.countDoubleSharing(hostelId), roomRepository.countTripleSharing(hostelId),
+				roomRepository.countMiscSharing(hostelId));
+	}
+
+	@GetMapping("hostels/{id}/getName")
```
Comment (srans-admin) [inline src/main/java/com/srans/nestserver/controller/HostelController.java:278]: getName case should be follow across all APIs

File: src/main/java/com/srans/nestserver/controller/HostelController.java [hunk 21]
```diff
@@ -17,59 +22,271 @@
...
+
+	public Map<String, Object> getHostelMap(@PathVariable(value = "id") Long hostelId) {
+		Map<String, Object> map = new HashMap<>();
+		map.put("Total Floor : ", hostelRepository.numOfFloor(hostelId));
+		map.put("Hostel Name ", hostelRepository.hostelName(hostelId));
+		return map;
+	}
+
+	@GetMapping("hostels/{id}/hosteldetails")
```
Comment (srans-admin) [inline src/main/java/com/srans/nestserver/controller/HostelController.java:287]: hostel-details on API endpoint

File: src/main/java/com/srans/nestserver/controller/InvoiceController.java [hunk 22]
```diff
@@ -0,0 +1,77 @@
...
+import com.srans.nestserver.exception.ResourceNotFoundException;
+import com.srans.nestserver.model.Invoice;
+import com.srans.nestserver.repository.InvoiceRepository;
+
+@CrossOrigin(origins = "http://localhost:4200",allowedHeaders = "*")
+@RestController
+@RequestMapping("/api/v1")
+public class InvoiceController {
+	//Logger logger = LoggerFactory.getLogger(InvoiceController.class);
+	@Autowired
```
Comment (srans-admin) [inline src/main/java/com/srans/nestserver/controller/InvoiceController.java:32]: Implement Logging

File: src/main/java/com/srans/nestserver/controller/ManagementController.java [hunk 23]
```diff
@@ -0,0 +1,83 @@
...
+
+
+
+
+
+@CrossOrigin(origins = "http://localhost:4200",allowedHeaders = "*") 
+@RestController
+@RequestMapping("/api/v1")
+public class ManagementController {
+	//Logger logger = LoggerFactory.getLogger(ManagementController.class);
```
Comment (srans-admin) [inline src/main/java/com/srans/nestserver/controller/ManagementController.java:37]: Implement Logging

File: src/main/java/com/srans/nestserver/controller/ManagementController.java [hunk 24]
```diff
@@ -0,0 +1,83 @@
...
+	@Autowired
+	private ManagementRepository managementRepository;
+	
+	@GetMapping("/management")
```
Comment (srans-admin) [inline src/main/java/com/srans/nestserver/controller/ManagementController.java:41]: What is the meaning of management?

File: src/main/java/com/srans/nestserver/controller/RoomController.java [hunk 25]
```diff
@@ -0,0 +1,21 @@
...
+import org.springframework.web.bind.annotation.CrossOrigin;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+
+
+@CrossOrigin(origins = "http://localhost:4200", allowedHeaders = "*")
+@RequestMapping("api/v1")
+@RestController
+public class RoomController {
```
Comment (srans-admin) [inline src/main/java/com/srans/nestserver/controller/RoomController.java:13]: What is this empty Controller?

File: src/main/java/com/srans/nestserver/controller/TenantBookingController.java [hunk 26]
```diff
@@ -0,0 +1,47 @@
...
+@RequestMapping("/api/v1")
+public class TenantBookingController {
+
+	Logger logger = LoggerFactory.getLogger(TenantController.class);
+
+	@Autowired
+	private TenantRepository tenantRepository;
+
+	
+	@GetMapping("hostels/hostelName")
```
Comment (srans-admin) [inline src/main/java/com/srans/nestserver/controller/TenantBookingController.java:33]: Wht this API, should be design correctly

File: src/main/java/com/srans/nestserver/controller/TenantBookingController.java [hunk 27]
```diff
@@ -0,0 +1,47 @@
...
+	public List<String> findAll() {
+		return tenantRepository.getAllHostelName();
+	}
+
+	@GetMapping(value="hostels/{hostelname}/bed", produces = MediaType.APPLICATION_JSON_VALUE)
```
Comment (srans-admin) [inline src/main/java/com/srans/nestserver/controller/TenantBookingController.java]: hostelname -> hostelName

File: src/main/java/com/srans/nestserver/controller/TenantController.java [hunk 28]
```diff
@@ -0,0 +1,163 @@
...
+	private TenantBookRepository tenantBookRepository;
+	
+	
+@PostMapping("/tenant")
+	
+	public Tenant saveTenant(@Valid @RequestBody Tenant tenant) throws NSException {
+		
+		logger.info("IN::POST::/hostels::saveHostel::" + tenant);
+
+		Tenant responsetenant=tenantRepository.save(tenant);
```
Comment (srans-admin) [inline src/main/java/com/srans/nestserver/controller/TenantController.java:61]: tenantbooking -> tenantBooking

File: src/main/java/com/srans/nestserver/controller/TenantController.java [hunk 29]
```diff
@@ -0,0 +1,163 @@
...
+	@GetMapping("/tenantidproof/{id}/retrive/{cat}")
+	public ResponseEntity<InputStreamResource> retriveIdproofImage(@PathVariable("id") Long id,
+			@PathVariable("cat") String cat) throws NSException, IOException {
+
+		return ResponseEntity.ok().contentType(MediaType.IMAGE_PNG)
+				.body(new InputStreamResource(storageService.retriveIdproofImage(id, cat)));
+
+	}
+
+	@PostMapping("/tenantidproof/{id}/upload/{cat}")
```
Comment (srans-admin) [inline src/main/java/com/srans/nestserver/controller/TenantController.java:116]: use proper naming convention with correct API design

File: src/main/java/com/srans/nestserver/controller/TenantController.java [hunk 30]
```diff
@@ -0,0 +1,163 @@
...
+
+	}
+
+	/*
+	 * @PostMapping("/tenant") public Tenant createUser(@RequestBody Tenant tenant)
+	 * { System.out.println("User : " + tenant); return
+	 * tenantRepository.save(tenant); }
+	 */
+
+	@PutMapping("/tenant/{Id}")
```
Comment (srans-admin) [inline src/main/java/com/srans/nestserver/controller/TenantController.java:132]: Id -> id

File: src/main/java/com/srans/nestserver/exception/MyFileNotFoundException.java [hunk 31]
```diff
@@ -0,0 +1,5 @@
+package com.srans.nestserver.exception;
+
+public class MyFileNotFoundException {
```
Comment (srans-admin) [inline src/main/java/com/srans/nestserver/exception/MyFileNotFoundException.java]: why this class?

**Review by srans-admin — CHANGES_REQUESTED** [review]
Very overal, PR not at all looks good. Please try to follow my instructions and re-implement the things.
You need to learn following first before implement it looks like
1. Java naming conventions
2. Documentation comments
3. API Design standards
4. /target folder should not commit to GIT

File: src/main/java/com/srans/nestserver/controller/ChartController.java (hunk 5 above)
Comment (Manish-Srans) [reply src/main/java/com/srans/nestserver/controller/ChartController.java]: OK Sir...

**Comment by Manish-Srans** [comment]
Sir, Please Check Again My PR

**Comment by srans-admin** [comment]
Closing this PR as he raised another equivalent PR with requested changes accommodated

**Pull request closed without merging**